espresso-tests/smoke-test-force-inclusion-test.bash

```diff
@@ -0,0 +1,231 @@
+CAFF_NODE_RESPONSE=$(cast balance 0x3f1Eae7D46d88F08fc2F8ed27FCb2AB183EB2d0E --rpc-url http://127.0.0.1:8550)
+
+if [[ $CAFF_NODE_RESPONSE == "0" ]]; then
+    echo "Caff node is not running. Please start the node and try again."
+    exit 1
+fi
+
+
+PRIVATE_KEY="$(docker compose run scripts print-private-key --account l2owner 2>/dev/null | trim-last)"
+# This is a private key used for testing, save to print
+declare -p PRIVATE_KEY
+
+# Set the max delay blocks to 1 block, future blocks to 120 blocks, delay seconds to 360 seconds, future seconds to 3600 seconds
+cast send $PARENT_CHAIN_UPGRADE_EXECUTOR $(cast calldata "executeCall(address, bytes)" $SEQUENCER_INBOX  $(cast calldata "setMaxTimeVariation((uint256,uint256,uint256,uint256))"  "(1,120,1,3600)"))  --rpc-url $PARENT_CHAIN_RPC_URL --private-key $PRIVATE_KEY
+
+#  First call the maxTimeVariation function to get the max time variation
+{
+  read DELAY_BLOCKS
+  read FUTURE_BLOCKS
+  read DELAY_SECONDS
+  read FUTURE_SECONDS
+} < <(
+  cast call --rpc-url "$PARENT_CHAIN_RPC_URL" "$SEQUENCER_INBOX" \
+    'maxTimeVariation()(uint256,uint256,uint256,uint256)' |
+    awk '{print $1}'  # Extract only the first field (in case of extra text like "[8.64e4]")
+)
+
+echo "DELAY_BLOCKS: $DELAY_BLOCKS"
+echo "FUTURE_BLOCKS: $FUTURE_BLOCKS"
+echo "DELAY_SECONDS: $DELAY_SECONDS"
+echo "FUTURE_SECONDS: $FUTURE_SECONDS"
+
+# Now we stop the sequencer 
+run docker stop nitro-testnode-sequencer-1
+
+USER_L1_PRIVATE_KEY="$(docker compose run scripts print-private-key --account funnel 2>/dev/null | trim-last)"
+# This is a private key used for testing, save to print
+declare -p USER_L1_PRIVATE_KEY
+
+# Add retry logic to send delayed message because sometimes it fails on the first attempt
+MAX_RETRIES=5
+RETRY_DELAY=5  # seconds between retries
+
+# Your original command
+CMD="cast send $INBOX_ADDRESS \$(cast calldata \"sendL2MessageFromOrigin(bytes)\" \"0x123456\") --rpc-url $PARENT_CHAIN_RPC_URL --private-key $USER_L1_PRIVATE_KEY --gas-limit 20000000"
+
+# Retry logic
+retry_count=0
+while [ $retry_count -lt $MAX_RETRIES ]; do
+    echo "Attempt $((retry_count + 1)) of $MAX_RETRIES..."
+    
+    if eval "$CMD"; then
+        echo "Command succeeded!"
+        break
+    else
+        echo "Command failed. Retrying in $RETRY_DELAY seconds..."
+        sleep $RETRY_DELAY
+        ((retry_count++))
+    fi
+done
+
+if [ $retry_count -ge $MAX_RETRIES ]; then
+    echo "All retries failed. Exiting."
+    exit 1
+fi
+
+sleep 120
+
+
+CAFF_NODE_RESPONSE=$(cast balance 0x3f1Eae7D46d88F08fc2F8ed27FCb2AB183EB2d0E --rpc-url http://127.0.0.1:8550 2>/dev/null)
+
+# Check if the command succeeded (got a balance response)
+if [[ $? -eq 0 ]]; then
+    # If we got a balance response (success case for the command), that's bad for us
+    if [[ $CAFF_NODE_RESPONSE != "0" ]]; then
+        echo "Caff node should not be running - balance is $CAFF_NODE_RESPONSE"
+        exit 1
+    fi
+fi
+echo "Test Passed"
```

Comment:
Would it be better for us to check the logs of the caff node container? We can add it after we have this tool

Reply:
I will add it!